src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -2552,12 +2552,13 @@ void java_lang_Throwable::fill_in_stack_trace(Handle throwable, const methodHand
     return;
   }
 
-  PRESERVE_EXCEPTION_MARK;
+  PreserveExceptionMark pm(Thread::current()); // not necessarily JavaThread::active().
 
   JavaThread* thread = JavaThread::active();
```

Comment:
This should NOT be JavaThread::active() since it's passed and used as TRAPS, ie, as the current thread for creating ResourceMark and throwing exceptions.  This is wrong.